internal/pkg/manifest/validate.go

```diff
@@ -115,6 +115,24 @@ func (d DeploymentConfiguration) validate() error {
 	return fmt.Errorf("invalid rolling deployment strategy %s, must be one of %s",
 		aws.StringValue(d.Rolling),
 		english.WordSeries(ecsRollingUpdateStrategies, "or"))
+	if err := d.Alarms.validate(); err != nil {
+		return fmt.Errorf("validate rollback alarms: %w", err)
+	}
+	return nil
+}
+
+func (r RollbackAlarmArgsOrNames) validate() error {
+	if r.AlarmArgs.isEmpty() {
+		return nil
+	}
```

Comment:
nit: I feel like this block could be written in L134 in `func (a AlarmArgs) validate() errror`  - my reasoning is that whether being empty is valid or not sound like a judgement that `AlarmArgs` should make.